### PR TITLE
chore: Add Prettier and ESLint to the CCIP Gateway package

### LINKED ISSUE #280: CCIP Gateway: improve developer experience
- [ ] Update dev/test dependencies
- [ ] Add eslint
- [ ] Add prettier
- [ ] Fix all linting issues
- [ ] Fix all typing issues

### PR DESCRIPTION
**Pull request by alainncls**
Fixes #280 

This PR improves the developer experience on the `linea-ccip-gateway` package, by adding:
- prettier
- eslint

### CODE REVIEW

File: packages/linea-ccip-gateway/package.json [hunk 1]
```diff
@@ -1,56 +1,59 @@
...
     "dotenv": "^16.0.3",
     "ethers": "^6.11.1",
     "express": "^4.19.2",
-    "@consensys/linea-state-verifier": "^1.0.1",
     "tsdx": "^0.14.1",
     "tslib": "^2.6.2",
-    "winston": "^3.9.0",
-    "typescript": "^5.4.5"
+    "typescript": "^5.4.5",
+    "winston": "^3.9.0"
```
Comment (thedarkjester) [inline packages/linea-ccip-gateway/package.json:30]: while you are here, I would specify/fix the versions

File: packages/linea-ccip-gateway/src/evm-gateway/EVMGateway.ts [hunk 2]
```diff
@@ -125,13 +125,13 @@ export class EVMGateway<T extends ProvableBlock> {
   /**
    *
    * @param address The address to fetch storage slot proofs for
-   * @param paths Each element of this array specifies a Solidity-style path derivation for a storage slot ID.
-   *              See README.md for details of the encoding.
+   * @param commands The commands to execute to compute the storage slots
+   * @param constants The constants to use in the commands
```
Comment (thedarkjester) [inline packages/linea-ccip-gateway/src/evm-gateway/EVMGateway.ts:129]: the comments before seemed relevant in terms of encoding? still applicable?

File: packages/linea-ccip-gateway/src/server.ts [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 const l1ProviderUrl = process.env.L1_PROVIDER_URL;
 const l2ProviderUrl = process.env.L2_PROVIDER_URL;
-const l2ChainId = parseInt(process.env.L2_CHAIN_ID);
-const rollupAddress = process.env.L1_ROLLUP_ADDRESS;
+const l2ChainId = parseInt(process.env.L2_CHAIN_ID ?? '59144');
+const rollupAddress =
+  process.env.L1_ROLLUP_ADDRESS ?? '0xB218f8A4Bc926cF1cA7b3423c154a0D627Bdb7E5';
 const port = process.env.PORT || 3000;
-const nodeEnv = process.env.NODE_ENV || "test";
+const nodeEnv = process.env.NODE_ENV || 'test';
```
Comment (thedarkjester) [inline packages/linea-ccip-gateway/src/server.ts:15]: is this meant to default to mainnet given the example was changed to testnet? 59144 vs 59141

File: packages/linea-ccip-gateway/test/testVerifier.spec.ts [hunk 4]
```diff
@@ -1,36 +1,36 @@
...
-import { expect } from "chai";
+import { Server } from '@chainlink/ccip-read-server';
+import { makeL2Gateway } from '../src';
+import { HardhatEthersProvider } from '@nomicfoundation/hardhat-ethers/internal/hardhat-ethers-provider';
+import { HardhatEthersHelpers } from '@nomicfoundation/hardhat-ethers/types';
+import { expect } from 'chai';
 import {
   BrowserProvider,
   Contract,
+  ethers as ethersT,
```
Comment (thedarkjester) [inline packages/linea-ccip-gateway/test/testVerifier.spec.ts:9]: ethersT doesn't feel like a good name, I would have renamed the other one to ethersHardhat

File: packages/linea-ccip-gateway/src/server.ts (hunk 3 above)
Comment (alainncls) [reply packages/linea-ccip-gateway/src/server.ts:15]: Yeah, it's probably better to set it to `59141`!

File: packages/linea-ccip-gateway/src/evm-gateway/EVMGateway.ts (hunk 2 above)
Comment (alainncls) [reply packages/linea-ccip-gateway/src/evm-gateway/EVMGateway.ts:129]: I don't see anything related to encoding in the README file, so I don't think this is relevant anymore

File: packages/linea-ccip-gateway/package.json (hunk 1 above)
Comment (alainncls) [reply packages/linea-ccip-gateway/package.json]: Still, I upgraded a few dependencies to improve the test environment.

File: packages/linea-ccip-gateway/test/testVerifier.spec.ts [hunk 5]
```diff
@@ -39,158 +39,151 @@ describe("L1Verifier", () => {
...
-    l1SepoliaProvider = new ethers.JsonRpcProvider(
-      "https://gateway.tenderly.co/public/sepolia",
+    l1SepoliaProvider = new ethersHardhat.JsonRpcProvider(
+      'https://gateway.tenderly.co/public/sepolia',
       11155111,
       {
         staticNetwork: true,
-      }
+      },
     );
```
[the remark this answers is not in the document]
Comment (alainncls) [reply packages/linea-ccip-gateway/test/testVerifier.spec.ts]: Done in `packages/linea-ccip-gateway/test/constants.ts`! 🫡

**Review by thedarkjester — APPROVED** [review]

**Pull request merged**